tests/smoke.rs

```diff
@@ -445,6 +468,30 @@ fn dependencies() {
     run!()
 }
 
+/// Test that `xargo build` can be executed from outside project dir
+#[cfg(feature = "dev")]
+#[test]
+fn build_outside_project_dir() {
+    fn run() -> Result<()> {
+        // need this exact target name to get the right gcc flags
+        const TARGET: &'static str = "thumbv6m-workdir-eabi";
+
+        let project = Project::new(TARGET)?;
+        project.xargo_toml(
+            r#"
+[dependencies.alloc]
+"#,
+        )?;
+        let td = TempDir::new("xargo_workdir").chain_err(|| "couldn't create a temporary directory")?;
```

Comment:
So crucially, this is a `TempDir` that is *separate* from the `TempDir` inside `project`, right? If so, please add a comment explaining that.

Reply:
correct.fixed.